File: CiaiControllerSDKForJava/src/main/java/com/ciai/controller/sdk/interface_/ICommunicationProvider.java
package com.ciai.controller.sdk.interface_;

import com.ciai.controller.sdk.core.ConnectionConfiguration;
import java.util.Collection;

/** 厂商协议或第三方通信实现的注册入口。 */
public interface ICommunicationProvider {
    Collection<String> getTypes();
    void validate(ConnectionConfiguration configuration);
    ICommunication create(ConnectionConfiguration configuration);
}

File: CiaiControllerSDK/Interfaces/ICommunicationProvider.cs
using System.Collections.Generic;
using CiaiControllerSDK.Core;

namespace CiaiControllerSDK.Interfaces
{
    /// <summary>厂商协议或第三方通信实现的注册入口，无需修改SDK工厂。</summary>
    public interface ICommunicationProvider
    {
        IEnumerable<string> Types { get; }
        void Validate(ConnectionConfiguration configuration);
        ICommunication Create(ConnectionConfiguration configuration);
    }
}
